client/driver/env/env.go

```diff
@@ -103,8 +103,11 @@ func (t *TaskEnvironment) Build() *TaskEnvironment {
 	}
 
 	// Build the ports
-	for label, port := range t.ports {
-		t.taskEnv[fmt.Sprintf("%s%s", PortPrefix, label)] = strconv.Itoa(port)
+	for _, network := range t.networks {
+		for label, value := range network.MapLabelToValues(t.portMap) {
```

Comment:
Not sure if you want to do it in this PR or a follow up one but we should add a `$NOMAD_HOST_PORT_"label"=port` if there exists a port map for a label. There are certain applications that need to know the actual port